attribute-masking-service/README.md

```diff
@@ -35,37 +39,37 @@ The service performs two functions:
 | user                    | maskedLeafResource       | exception         | user
 | leafResource            |                          | serverMetadata    | leafResource
 | rules                   |                          |                   | rules
-(fields marked with * are acquired from headers metadata)
 
-The service takes in an `AttributeMaskingRequest` and a request `Token`.
-This request is persisted in a store as an `AuthorisedRequestEntity`.
-The `LeafResource` attached to the request is then masked, removing any excessive or sensitive metadata which a client may be prohibited from accessing.
-The resulting masked `LeafResource` is then attached to an `AttributeMaskingResponse` and outputted from the service.
-
-### Future Enhancements
-The masking operation may in the future apply attribute-level `Rule`s using the `User` and `Context` for fine-grained control over client metadata access.
+*token is in the message header's metadata
 
+The service reads a message in from the Kafka "rule" topic as a [AttributeMaskingRequest](src/main/java/uk/gov/gchq/palisade/service/attributemask/model/AttributeMaskingRequest.java).
+This request is persisted in a store as an`AuthorisedRequestEntity`.
+The `LeafResource` attached to the request is then masked, removing any excessive or sensitive metadata which a client may be prohibited from accessing.
+The resulting masked `LeafResource` is then used to create an [AttributeMaskingResponse](src/main/java/uk/gov/gchq/palisade/service/attributemask/model/AuditableAttributeMaskingResponse.java) ready to be forwarded.
+This will be forwarded onto the Kafka topic "masked-resource".
+Start and end markers are forwarded to the same Kafka topic.
+Like the masked resource messages, the start and end markers will have the token (`x-request-token`) in the header, but will have an empty message body.
+Errors that occur during the process are bundled into an error message, `AuditErrorMessage` and sent to the Kafka topic "error" where they are retrieved by the Audit Service.
```

Comment:
```suggestion
Any errors that occur when processing a message create an `AuditErrorMessage`, which is added to the `error` Kafka topic where they are processed by the Audit Service.
```

Reply:
Picked some of the changes and added a few of my own.  Have another look.